### PR TITLE
ENG-510: link lens profile to ik profile

### PR DESCRIPTION
**Pull request by stormcloud266**
- add lens hooks, get lens info in profile
- add user update on login

Screenshot 2023-04-13 at 12 15 43 PM

### CODE REVIEW

File: web/src/components/LensConnect/LensConnect.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      })
+    }
+  }, [hasTriedConnection, profile, create])
+
+  const onLoginClick = async () => {
+    try {
+      const signer = await connector.getSigner()
+      await connectToLens(signer)
+    } catch {
+      console.error('Error connecting to Lens Profile')
```
Comment (illepic) [inline web/src/components/LensConnect/LensConnect.tsx:61]: Anything we can show in the UI here?

File: web/src/components/LensConnect/LensConnect.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    }
+    setHasTriedConnection(true)
+  }
+
+  const onLogoutClick = () => {
+    create({
+      variables: {
+        input: {
+          lensProfile: null,
```
Comment (illepic) [inline web/src/components/LensConnect/LensConnect.tsx:70]: Double checking here: is this removing lens profile data from the user's profile on lens logout?

File: web/src/components/LensConnect/LensConnect.tsx (hunk 1 above)
Comment (illepic) [reply web/src/components/LensConnect/LensConnect.tsx:61]: Or rather, will the `connectToLensError` status be set here regardless? (which i see we handle below)

**Review by illepic — CHANGES_REQUESTED** [review]

File: web/src/components/LensConnect/LensConnect.tsx (hunk 2 above)
Comment (stormcloud266) [reply web/src/components/LensConnect/LensConnect.tsx:70]: yes it removes their lens handle from their user data when they manually disconnect from lens

File: web/src/components/LensConnect/LensConnect.tsx (hunk 1 above)
Comment (stormcloud266) [reply web/src/components/LensConnect/LensConnect.tsx:61]: Right the `connectToLensError` gets set automatically

**Review by illepic — APPROVED** [review]

**Pull request merged**